[package] libcurl/8.19.0: Windows static build with libnghttp2 via CMake/Conan links against __imp__nghttp2_version

### Description

Building libcurl with with_nghttp2=True and static libnghttp2 on Windows/MSVC produces a linker mismatch.
The resulting libcurl.lib(http2.obj) references __imp__nghttp2_version, as if nghttp2 were a DLL import, but libnghttp2 is static.
`libcurl.lib(http2.obj) : error LNK2001: unresolved external symbol __imp__nghttp2_version`

Older CCI recipe revisions patched curl’s CMake NGHTTP2 discovery to use Conan targets directly.
Newer recipe revisions rely on CMakeDeps aliasing / cmake_target_name.
This appears to regress static Windows handling.

*Observations*
`libnghttp2` exports `NGHTTP2_STATICLIB` in its package metadata.
The failure suggests that define is not reaching curl’s http2 compilation.
Setting `NGHTTP2_USE_STATIC_LIBS=TRUE` for curl’s CMake build or directly forcing `NGHTTP2_STATICLIB` appears to fix/work around it.

*Is the current libcurl CCI recipe expected to support static libnghttp2 on Windows with CMake/MSVC?
Should the recipe set NGHTTP2_USE_STATIC_LIBS when libnghttp2 is static?
Or should the recipe restore explicit CMake patching for NGHTTP2 discovery?*

### Package and Environment Details

* Package Name/Version: **libcurl/8.19.0**
* Operating System+version: **Windows**
* Compiler+version: **MSVC 19.29.30157.0**
* Docker image: **N/A**
* Conan version: **conan 1.53.0**
* Python version: **Python 3.9.23**
* libnghttp2 version: **1.68.1**
* libcurl / libnghttp2 **shared=False**

`def configure(self):
    self.options["libcurl/*"].with_nghttp2 = True
    self.options["libcurl/*"].shared = False
    self.options["libnghttp2/*"].shared = False`


### Conan profile

[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=192
os=Windows


### Steps to reproduce

conan install . -pr <profile> -pr:b build-win-x86_64
cmake --preset default
cmake --build --preset release

### Logs

N/A

--- a/recipes/libcurl/all/conanfile.py
+++ b/recipes/libcurl/all/conanfile.py
@@ -573,6 +573,8 @@ class LibcurlConan(ConanFile):
             tc.variables["CURL_DISABLE_WEBSOCKETS"] = not self.options.with_websockets
         if self.options.with_nghttp2:
             tc.variables["NGHTTP2_USE_STATIC_LIBS"] = not self.dependencies["libnghttp2"].options.shared
+        if self.options.with_zstd:
+            tc.variables["ZSTD_USE_STATIC_LIBS"] = not self.dependencies["zstd"].options.shared
 
         # Also disables NTLM_WB if set to false
         if not self.options.with_ntlm:

--- a/recipes/libcurl/all/conanfile.py
+++ b/recipes/libcurl/all/conanfile.py
@@ -220,6 +220,13 @@ class LibcurlConan(ConanFile):
         cert_url = self.conf.get("user.libcurl.cert:url", check_type=str) or "https://curl.se/ca/cacert-2025-11-04.pem"
         cert_sha256 = self.conf.get("user.libcurl.cert:sha256", check_type=str) or "8ac40bdd3d3e151a6b4078d2b2029796e8f843e3f86fbf2adbc4dd9f05e79def"
         download(self, cert_url, "cacert.pem", verify=True, sha256=cert_sha256)
+        replace_in_file(self, "CMakeLists.txt", "find_package(NGHTTP2 MODULE)", "find_package(NGHTTP2 CONFIG REQUIRED)")
+        replace_in_file(self, "CMakeLists.txt", "find_package(Cares MODULE REQUIRED)", "find_package(Cares CONFIG REQUIRED)")
+        replace_in_file(self, os.path.join("CMake", "Macros.cmake"), "find_package(${_find_name})", "find_package(${_find_name} CONFIG REQUIRED)")
+        replace_in_file(self, os.path.join("CMake", "Macros.cmake"), "find_package(${_find_name} MODULE)", "find_package(${_find_name} CONFIG REQUIRED)")
+        replace_in_file(self, os.path.join("CMake", "Macros.cmake"), "find_package(${_find_name} REQUIRED)", "find_package(${_find_name} CONFIG REQUIRED)")
+        replace_in_file(self, os.path.join("CMake", "Macros.cmake"), "find_package(${_find_name} MODULE REQUIRED)", "find_package(${_find_name} CONFIG REQUIRED)")
+
 
     def generate(self):
         env = VirtualBuildEnv(self)
@@ -571,10 +578,6 @@ class LibcurlConan(ConanFile):
             tc.variables["CURL_DISABLE_FORM_API"] = not self.options.with_form_api
         if "with_websockets" in self.options:
             tc.variables["CURL_DISABLE_WEBSOCKETS"] = not self.options.with_websockets
-        if self.options.with_nghttp2:
-            tc.variables["NGHTTP2_USE_STATIC_LIBS"] = not self.dependencies["libnghttp2"].options.shared
-        if self.options.with_zstd:
-            tc.variables["ZSTD_USE_STATIC_LIBS"] = not self.dependencies["zstd"].options.shared
 
         # Also disables NTLM_WB if set to false
         if not self.options.with_ntlm:

--- a/recipes/libcurl/all/conanfile.py
+++ b/recipes/libcurl/all/conanfile.py
@@ -571,6 +571,8 @@ class LibcurlConan(ConanFile):
             tc.variables["CURL_DISABLE_FORM_API"] = not self.options.with_form_api
         if "with_websockets" in self.options:
             tc.variables["CURL_DISABLE_WEBSOCKETS"] = not self.options.with_websockets
+        if self.options.with_nghttp2 and is_msvc(self) and not self.dependencies["libnghttp2"].options.shared:
+            tc.preprocessor_definitions["NGHTTP2_STATICLIB"] = "1"
 
         # Also disables NTLM_WB if set to false
         if not self.options.with_ntlm:

--- a/recipes/libcurl/all/conanfile.py
+++ b/recipes/libcurl/all/conanfile.py
@@ -571,8 +571,8 @@ class LibcurlConan(ConanFile):
             tc.variables["CURL_DISABLE_FORM_API"] = not self.options.with_form_api
         if "with_websockets" in self.options:
             tc.variables["CURL_DISABLE_WEBSOCKETS"] = not self.options.with_websockets
-        if self.options.with_nghttp2 and is_msvc(self) and not self.dependencies["libnghttp2"].options.shared:
-            tc.preprocessor_definitions["NGHTTP2_STATICLIB"] = "1"
+        if self.options.with_nghttp2:
+            tc.variables["NGHTTP2_USE_STATIC_LIBS"] = not self.dependencies["libnghttp2"].options.shared
 
         # Also disables NTLM_WB if set to false
         if not self.options.with_ntlm:

